src/main/java/io/outbound/sdk/RequestHandler.java

```diff
@@ -64,6 +64,22 @@ public void processAfterDelay(OutboundRequest request, long delayMillis) {
         handler.postDelayed(buildRunnableRequest(request), delayMillis);
     }
 
+    public void processImmediately(OutboundRequest request) {
+        ensureInitialized();
+
+        if (canProcess()) {
+            Status immediateSendStatus = sendQueuedRequest(request);
```

Comment:
We should try more than once here before queuing.

Reply:
I thought about this a bit more, I think I might change it so that if it fails it then reschedules the Firebase job rather than using queue.